File: dual_trigger/dual.py
def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        l = input()
        num_ones = 0
        for elem in l:
            if elem == '1':
                num_ones += 1
        print("NO") if num_ones % 2 or (num_ones == 2 and "11" in l) else print("YES")
    
    
if __name__ == "__main__":
    main()

File: dual_trigger/dual.cpp
#include <iostream>
#include <string>

using namespace std;

int main(void)
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n;
        cin >> n;
        string l;
        cin >> l;
        int num_ones = 0;
        for (int j = 0; j < n; j++)
        {
            if (l[j] == '1')
            {
                num_ones++;
            }
        }
        string ans = (num_ones % 2 || (num_ones == 2 && (l.find("11") != string::npos))) ? "NO\n" : "YES\n";
        cout << ans;
    }
}
